std/net/punycode.d

```diff
@@ -0,0 +1,232 @@
+/**
+	Encodes and decodes punycode strings, per
+	$(LINK2 https://www.ietf.org/rfc/rfc3492.txt, RFC3492).
+	Punycode is primarily used for converting URIs or IRIs containing non-ASCII
+	characters into URIs or IRIs that are entirely ASCII, and vice versa.
+	This punycode codec is based upon the original implementation found in
+	$(LINK2 https://www.ietf.org/rfc/rfc3492.txt, RFC3492).
+
+	Authors: Shotaro Yamada (Sinkuu)
+	Date: February 27, 2017
+	License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+	Standards: $(LINK2 https://www.ietf.org/rfc/rfc3492.txt, RFC3492)
+	Source: $(PHOBOSSRC std/net/_punycode.d)
+	Version: 1.0.1
+*/
+module punycode;
+
+private import std.ascii : isASCII, isUpper, isLower, isDigit;
+private import std.conv : to;
+private import std.exception : enforce;
+private import std.traits : isSomeString;
+private import std.array : insertInPlace;
+private import std.algorithm.searching : all;
+
+version (unittest)
+{
+	import std.exception : assertThrown, collectExceptionMsg;
+}
+
+private immutable uint base = 36;
+private immutable ubyte initialN = 0x80;
+private immutable uint initialBias = 72;
+private immutable uint tmin = 1;
+private immutable uint tmax = 26;
+private immutable uint damp = 700;
+private immutable uint skew = 38;
+
+/**
+	Converts an UTF string to a Punycode string.
+	Params: str = A UTF-encoded string that should be encoded into Punycode
+	Returns: A punycode-encoded string
+	Throws: PunycodeException if an internal error occured
+*/
+S punyEncode(S)(in S str) @safe pure
```

Comment:
I thought Walter had implemented it? It's better to leave it as `const` anyway because even if the changes to scope have not yet been implemented, this may break in weird ways when it _does_ get implemented.